src/myrtlespeech/data/sampler.py

```diff
@@ -1,14 +1,47 @@
 import random
 
 
-class RandomBatchSampler:
-    """TODO"""
+class SequentialRandomSampler:
+    """A sequential or random iterable over batches.
 
-    def __init__(self, indices, batch_size, shuffle, drop_last=False):
+    The iterator used each time this iterable is iterated over will yield
+    batches either sequentially (i.e. in-order) or randomly (uniform without
+    replacement) from `batches`.
+    This iterable records the number of times it has returned an iterator. A
+    sequential iterator is returned if the current count is in `sequential`.
+
+    Args:
+        indices: data with which batches are created.
+        batch_size: Batch dimension.
+        shuffle: Set to True to have the data reshuffled at every epoch if a
+            random iterator is used.
+        drop_last: (optional, bool): Set to True to drop the last incomplete
+            batch, if the dataset size is not divisible by the batch size. If
+            False and the size of dataset is not divisible by the batch size,
+            then the last batch will be smaller.
+        n_iterators (optional, int): Number of iterators returned so far.
+        sequential (optional, set of int): Counts at which to return a
+            sequential iterator.
+
+    Yields:
+        Batches from `batches`.
+    """
+
+    def __init__(
+        self,
+        indices,
+        batch_size,
+        shuffle,
+        drop_last=False,
+        n_iterators=0,
+        sequential=None,
```

Comment:
The parameters should be typed and then the types in the docstring can be removed.